target/linux/mpc85xx/files/arch/powerpc/boot/dts/ecw7210.dts

```diff
@@ -0,0 +1,304 @@
+				reg = <0x1f40000 0x20000>;
+			};
+
+			partition@1f20000 {
+				read-only;
+				label = "mtd-mfr";
+				reg = <0x1f20000 0x20000>;
+			};
+
+			partition@1e20000 {
+				read-only;
+				label = "mtd-reserved";
+				reg = <0x1e20000 0x100000>;
+			};
+
+			partition@1e00000 {
+				read-only;
+				label = "crashlog";
+				reg = <0x1e00000 0x20000>;
+			};
+
+			partition@60000 {
+				read-only;
+				label = "mtd-unused";
+				reg = <0x60000 0x1da0000>;
+			};
+
+			partition@20000 {
+				read-only;
+				label = "mtd-dtb";
+				reg = <0x20000 0x40000>;
+			};
+
+			partition@0 {
+				read-only;
+				label = "mtd-u-script";
+				reg = <0x0 0x20000>;
+			};
+		};
+	};
+
+	board_soc: soc: soc@ffe00000 {
+		ranges = <0x0 0x0 0xffe00000 0x100000>;
+
+		i2c@3100 {
+			status = "okay";
+		};
+
+                mdio@25000 {
+			compatible = "fsl,etsec2-tbi";
+			tbi_phy1: tbi-phy@12 {
+				reg = <0x12>;
+			};	
+		};
+
+		
+		// We seem to need to have the tbi_phy's laid out this way for enet2 to work
+		mdio@24000 {
+			compatible = "fsl,etsec2-mdio";
+
+			phy0: ethernet-phy@0 {
+				reg = <0x0>;
+			};
+
+			phy1: ethernet-phy@1 {
+				reg = <0x1>;
+			};
+
+			tbi_phy0: tbi-phy@11 {
+				reg = <0x11>;
+			};
+		};
+
+
+		
+                eth0: enet2: ethernet@b2000 {
+			status = "okay";
+			phy-connection-type = "rgmii-id";
+			model = "eTSEC";
+			phy-handle = <&phy1>;
+			tbi-handle = <&tbi_phy1>;
+			local-mac-address = [70 72 cf 42 42 03];		
+		};
+		
+		eth1: enet0: ethernet@b0000 {
+			//fsl-gianfar
+			status = "okay";
+			phy-connection-type = "rgmii-id";
+			model = "eTSEC";
+			phy-handle = <&phy0>;
+			tbi-handle = <&tbi_phy0>;
+			local-mac-address = [70 72 cf 42 42 01];			
+		};
+
+		enet1: ethernet@b1000 {
+			status = "disabled";
+		};
+		
+	
+
+		gpio0: gpio-controller@fc00 {
+		};
+
+		usb@22000 {
+			phy_type = "ulpi";
+			dr_mode = "host";
+		};
+
+		usb@23000 {
+			status = "disabled";
+		};
+	};
+	
+	// Various fixups for things like interrupts and frequencies. U-boot usually does this,
+	// but it looks like the older u-boot doesn't like this newer device tree.
+	pci0: pcie@ffe09000 {
+		clock-frequency = <33333333>;
+		reg = <0x0 0xffe09000 0x0 0x1000>;
+		ranges = <0x2000000 0x0 0xa0000000 0x0 0xa0000000 0x0 0x20000000
+			0x1000000 0x0 0x00000000 0x0 0xffc30000 0x0 0x10000>;
+		pcie@0 {
+			ranges = <0x2000000 0x0 0xa0000000
+				  0x2000000 0x0 0xa0000000
+				  0x0 0x20000000
+
+				  0x1000000 0x0 0x0
+				  0x1000000 0x0 0x0
+				  0x0 0x100000>;
+		};
+
+
+	};
+
+	pci1: pcie@ffe0a000 {
+		reg = <0x0 0xffe0a000 0x0 0x1000>;
+		ranges = <0x2000000 0x0 0xc0000000 0x0 0xc0000000 0x0 0x20000000
+		      0x1000000 0x0 0x00000000 0x0 0xffc20000 0x0 0x10000>;
+		pcie@0 {
+			ranges = <0x2000000 0x0 0xc0000000
+			      0x2000000 0x0 0xc0000000
+				  0x0 0x20000000
+
+				  0x1000000 0x0 0x0
+				  0x1000000 0x0 0x0
+				  0x0 0x100000>;
+		};
+	};
+	
+
+};
+
+
+/include/ "fsl/p1020si-post.dtsi"
+
+
+
+/ {
```

Comment:
Why do you have a second root node here? Please tidy up your file before submitting it.

Reply:
Some of this is left-over from a week's worth of hacking. Don't get me wrong, I have some cleanup to do, and may have rushed this a bit to get it in before work started back.

That said, some of this second node may be necessary: I'd prefer to include fsl/p1020si-post.dtsi, which includes fsl/pq3-duart-0.dtsi, which then sets things like: 
`clock-frequency = <0>; `

My understanding is that u-boot is supposed to set that to a real value, but that doesn't happen with this device's u-boot. So, if I understand correctly, my options are:

1. Copy in fsl/p1020si-post.dtsi minus the include for fsl/pq3-duart-0.dtsi, and copy in aversion of fsl/pq3-duart-0.dtsi with the frequency set correctly.
2. Include fsl/p1020si-post.dtsi and then override it with a second node like this one to set the frequency.

I think approach 2 is cleaner, and looks like how a issue was handled in tl-wdr4900-v1.dts.

I might not be understanding correctly, and I will test it more when I do the rest of the .dts cleanup. In the least, I think I should narrow it down the the minimum changes needed after include of fsl/p1020si-post.dtsi (likely just the frequencies).

Let me know if that doesn't sound like a good approach.

Reply:
I did a _lot_ of experimenting, and got the .dts to be much cleaner. Thanks for pushing me to do so - see if it looks OK now.